src/client/shader.cpp

```diff
@@ -582,10 +586,10 @@ u32 ShaderSource::getShaderIdDirect(const std::string &name,
 	}
 
 	// Check if already have such instance
-	for(u32 i=0; i<m_shaderinfo_cache.size(); i++){
-		ShaderInfo *info = &m_shaderinfo_cache[i];
-		if(info->name == name && info->material_type == material_type &&
-			info->drawtype == drawtype)
+	for (u32 i = 0; i < m_shaderinfo_cache.size(); i++) {
+		auto &info = m_shaderinfo_cache[i];
+		if (info.name == name && info.base_material == base_mat &&
+			info.input_constants == input_const)
 			return i;
 	}
```

Comment:
I was worried if the material and string comparisons here might affect meshgen performance. But tested shortly with a pretty much empty mtg world, and it didn't appear in the flamegraph.

Reply:
BTW: I believe shaders are only retrieved once with very few exceptions.